### ISSUE TITLE
False positive for angular/di: [2, "array"] rule

### ISSUE THREAD

**Issue by reupen** [Gillespie59/eslint-plugin-angular#362]
In version 1.0.0, the following does not pass linting:
```
'use strict';

/*eslint angular/di: [2,"array"]*/

angular.module('myApp.myService')

    .factory('myService', ['myDep', myService]);

function myService(myDep) {
}
```

```
`$ ./node_modules/.bin/eslint test.js

test.js
  9:1  error  You should use the array syntax for DI  angular/di

✖ 1 problem (1 error, 0 warnings)
```

It was OK under 0.15.0. Also it passes under `/*eslint angular/di: [2,"function"]*/`.

**Comment by brutalcrozt** [comment]
Could you try use like so :
```
  angular.module('myApp.myService')
    .factory('myService', myService);
  myService.$inject = ['myDep'];
  function myService(myDep) {
  ...
  }
```

Let me know if it pass or even fail.

**Comment by reupen** [comment]
That (correctly) fails with `angular/di: [2, "array"]`

Yes, I would've expected the original example to pass under `array` and fail under `function`.

**Comment by brutalcrozt** [comment]
Sorry for that, i just notice that we have 3 possible configuration. 
When i compare your code and example. I found that your code use named function but the example use anonymous function.

According to the [test](https://github.com/Gillespie59/eslint-plugin-angular/blob/master/test/di.js#L83) not listed on valid one.

Let's wait the contributor :smile: , If require a PR maybe sorting the test by it's `configOption` will be more readable code ie :
``` 
{
code: ...,
options: ['array']
},{
code: ...,
options: ['array']
},{
code: ...,
options: ['fucntion']
},{
code: ...,
options: ['function']
},
etc


```

**Comment by Gillespie59** [comment]
I think with the "function" parameter, it should fail, because you use the array syntax. But should succeed with the "array" parameter. 

```
'use strict';

/*eslint angular/di: [2,"array"]*/

angular.module('myApp.myService')

    .factory('myService', ['myDep', myService]);

function myService(myDep) {
}
```

**Comment by Gillespie59** [comment]
I think the problem come from the way we detect AngularJS elements in our rules. can find for the moment a good approach in order to solve this issue :s
